lib/Object/ObjectLinker.cpp

```diff
@@ -3470,6 +3470,18 @@ bool ObjectLinker::readAndProcessInput(Input *Input, bool IsPostLto) {
     if (layoutInfo)
       layoutInfo->recordInputKind(CurInput->getKind());
     bool ELFOverriddenWithBC = false;
+    if (!isBackendInitialized()) {
+      // Infer machine for selecting backend
+      eld::Expected<uint16_t> Machine =
+          getELFExecObjParser()->getMachine(*CurInput);
```

Comment:
What do you think about adding the getMachine function to ELFReaderBase and moving this common code to outside input-kind if-condition?